objects/src/accounts/component/mod.rs

```diff
@@ -83,6 +89,33 @@ impl AccountComponent {
         Self::new(library, storage_slots)
     }
 
+    /// Instantiates an [AccountComponent] from the [ComponentPackage].
+    ///
+    /// The package's component metadata might contain templated values, which can be input by
+    /// mapping key names to [template values](TemplateValue) through the `template_keys`
+    /// parameter.
+    ///
+    /// # Errors
+    ///
+    /// - If any of the component's storage entries cannot be transformed into a valid storage slot.
+    ///   This could be because the metadata is invalid, or template values were not provided (or
+    ///   they are not of a valid type)
+    #[cfg(feature = "std")]
+    pub fn from_package(
+        package: &ComponentPackage,
+        template_keys: &BTreeMap<String, TemplateValue>,
+    ) -> Result<AccountComponent, ComponentPackageError> {
```

Comment:
We don't need the `std` restriction here, right?

Also, to carry over the comment from the other PR, this could look something like:
```rust
impl AccountComponent {

    pub fn from_package(
        package: &ComponentPackage,
        data: &InitStorageData,
    ) -> Result<Self, ComponentPackageError> {
        ...
    }

}
```
Lastly, I'm wondering if `ComponentPackage` is the best name here. To me, what we call "package" here is more like a "template". So, maybe it could be `AccountComponentTemplate` and then this method would be `from_template()`.

Reply:
Renamed this, going to work on `InitStorageData`

Reply:
As for the `std` restriction, it depends on how we want it I guess. Serialization/deserialization is based on `toml` and `serde` which is now gated by `std`. I can try to gate only serialization code and leave structs as they are right now.

Reply:
Yes, I think gating just the serialization/deserialization code is fine (we may have other ways to instantiate the structs in the future).